### PR TITLE
Improve filters

### PR DESCRIPTION
**Pull request by MohamedElmdary**
### Description

Describe the changes introduced by this PR and what does it affect

### Changes
Rewrote filters

### Related Issues

- #1869 
- #1901 
- #1902 
- #1957 
- #1970 

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/components/filters/TfFiltersContainer.vue [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+              <VRow no-gutters>
+                <slot />
+              </VRow>
+
+              <VRow class="mb-4" no-gutters v-show="valid && (changed || (!loading && !empty))">
+                <VAlert type="info" variant="tonal">
+                  <span>
+                    {{ changed ? "Filtering options updated but not applied." : "" }} Click
+                    <VCard class="d-inline pa-1" v-text="changed ? 'Apply' : 'Clear'" flat />
+                    {{ changed ? "inorder to reloading your data." : "to reset your selected filters." }}
```
Comment (mohamedamer453) [inline packages/playground/src/components/filters/TfFiltersContainer.vue:18]:
```suggestion
                    {{ changed ? "Filter options were updated but not applied." : "" }} Click
                    <VCard class="d-inline pa-1" v-text="changed ? 'Apply' : 'Clear'" flat />
                    {{ changed ? "in order to reload your data." : "to reset your selected filters." }}
```

File: packages/playground/src/dashboard/dedicated_nodes_view.vue [hunk 2]
```diff
@@ -89,56 +192,69 @@ const tabParams = {
...
+      page: page.value,
+      totalSru: convert(filters.value.minSSD),
+      totalMru: convert(filters.value.minRAM),
+      totalHru: convert(filters.value.minHDD),
+      totalCru: +filters.value.minCPU || undefined,
+      gpuVendorName: filters.value.gpuVendorName || undefined,
+      gpuDeviceName: filters.value.gpuDeviceName || undefined,
+      hasGpu: filters.value.gpu,
+    });
+
```
Comment (mohamedamer453) [inline packages/playground/src/dashboard/dedicated_nodes_view.vue:240]: remove console log if it's not needed.

**Review by mohamedamer453 — COMMENTED** [review]
- In  the country, region and farms filters i think it would look better if all countries, regions and farms are available by default instead of showing only the countries, regions and farms that are available at the moment.

**Comment by AlaaElattar** [comment]
- The highlight on Apply or Clear button in both dark and light modes, looks weird.
- Can we make it look better ?
image

**Comment by 0oM4R** [comment]
> * In  the country, region and farms filters i think it would look better if all countries, regions and farms are available by default instead of showing only the countries, regions and farms that are available at the moment.

the available country and region comes from all node status; up,standby and down. if the country is not listed on all of them why would we show it to the user? 
also the farm field aims to search and auto complete, is it really needed to load all farms ?

**Comment by mohamedamer453** [comment]
> the available country and region comes from all node status; up,standby and down. if the country is not listed on all of them why would we show it to the user?

If that's the case then i believe it's not working correctly, for example i'm running it locally on mainnet and in the country field when i search for egypt it doesn't show it.

image

even when there're nodes in egypt from the gridproxy. https://gridproxy.grid.tf/nodes?country=egypt

**Comment by mohamedamer453** [comment]
> also the farm field aims to search and auto complete, is it really needed to load all farms ?

for the farms, i didn't notice the `Load More Farms` button, i believe that's more than enough with the search/auto complete

**Comment by MohamedElmdary** [comment]
> * The highlight on Apply or Clear button in both dark and light modes, looks weird.
> * Can we make it look better ?
>   image

New Style

image
image

**Comment by MohamedElmdary** [comment]
> > the available country and region comes from all node status; up,standby and down. if the country is not listed on all of them why would we show it to the user?
> 
> If that's the case then i believe it's not working correctly, for example i'm running it locally on mainnet and in the country field when i search for egypt it doesn't show it.
> 
> image
> 
> even when there're nodes in egypt from the gridproxy. https://gridproxy.grid.tf/nodes?country=egypt

Fixed now location component reload all countries

**Review by mohamedamer453 — APPROVED** [review]

**Comment by Mahmoud-Emad** [comment]
Please fix the conflict

**Review by Mahmoud-Emad — APPROVED** [review]

**Pull request merged**